src/traces/scattergeo/hover.js

```diff
@@ -0,0 +1,105 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Fx = require('../../plots/cartesian/graph_interact');
+var Axes = require('../../plots/cartesian/axes');
+
+var getTraceColor = require('../scatter/get_trace_color');
+var attributes = require('./attributes');
+
+
+module.exports = function hoverPoints(pointData) {
+    var cd = pointData.cd,
+        trace = cd[0].trace,
+        xa = pointData.xa,
+        ya = pointData.ya,
+        geo = pointData.subplot;
+
+    if(cd[0].placeholder) return;
+
+    function c2p(lonlat) {
+        return geo.projection(lonlat);
+    }
+
+    function distFn(d) {
+        var lonlat = d.lonlat;
+
+        // this handles the not-found location feature case
+        if(lonlat[0] === null || lonlat[1] === null) return Infinity;
+
+        if(geo.isLonLatOverEdges(lonlat)) return Infinity;
+
+        var pos = c2p(lonlat);
+
+        var xPx = xa.c2p(),
+            yPx = ya.c2p();
+
+        var dx = Math.abs(xPx - pos[0]),
+            dy = Math.abs(yPx - pos[1]),
+            rad = Math.max(3, d.mrc || 0);
+
+        return Math.max(Math.sqrt(dx * dx + dy * dy) - rad, 1 - 3 / rad);
+    }
+
+    Fx.getClosest(cd, distFn, pointData);
+
+    // skip the rest (for this trace) if we didn't find a close point
+    if(pointData.index === false) return;
+
+    var di = cd[pointData.index],
+        lonlat = di.lonlat,
+        pos = c2p(lonlat),
+        rad = di.mrc || 1;
```

Comment:
is di.mrc undefined or 0? Or could be both and we want it to be 1 regardless. Also why 1? Maybe some comments here